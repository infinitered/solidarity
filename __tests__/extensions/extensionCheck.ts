import solidarityExtension from '../../src/extensions/solidarity-extension'
const context = require('mockContext')

const newContext = context
solidarityExtension(newContext)

test('Assure solidarity object', () => {
  expect(typeof newContext.solidarity).toBe('object')
})

test('Assure addPlugin function', () => {
  expect(typeof newContext.addPlugin).toBe('function')
})

test('Verify addPlugin function', () => {
  // plugin list is empty
  expect(newContext._pluginsList.length).toBe(0)
  newContext.addPlugin({})
  expect(newContext._pluginsList.length).toBe(1)
  // No existing path means empty template directory
  expect(newContext._pluginsList[0].templateDirectory).toBe(null)
  // TODO:  Create temporary templates folder and make sure it works
  // newContext.addPlugin({})
  // expect(newContext._pluginsList[0].templateDirectory).toBe(path.join(__dirname, '../templates/'))
})

test('Assure printSeparator function', () => {
  expect(typeof newContext.printSeparator).toBe('function')
})

test('Snapshot of solidarity extension', () => {
  expect(newContext.solidarity).toMatchSnapshot()
})
