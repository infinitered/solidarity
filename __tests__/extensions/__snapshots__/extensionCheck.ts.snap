// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot of solidarity extension 1`] = `
Object {
  "appendSolidaritySettings": [Function],
  "binaryExists": [Function],
  "buildSpecificRequirement": [Function],
  "checkCLI": [Function],
  "checkDir": [Function],
  "checkENV": [Function],
  "checkFile": [Function],
  "checkRequirement": [Function],
  "createPlugin": [Function],
  "getLineWithVersion": [Function],
  "getSolidaritySettings": [Function],
  "getVersion": [Function],
  "printResults": [Function],
  "removeNonVersionCharacters": [Function],
  "reviewRule": [Function],
  "ruleHandlers": Object {
    "cli": Object {
      "callback": [Function],
      "key": "binary",
    },
    "dir": Object {
      "callback": [Function],
      "key": "location",
    },
    "env": Object {
      "callback": [Function],
      "key": "variable",
    },
    "file": Object {
      "callback": [Function],
      "key": "location",
    },
    "shell": Object {
      "callback": [Function],
      "key": "command",
    },
  },
  "setOutputMode": [Function],
  "setSolidaritySettings": [Function],
  "skipRule": [Function],
  "updateRequirement": [Function],
  "updateVersions": [Function],
}
`;
